Update Python client CHANGELOG (#1437)
cmu-delphi/delphi-epidata: src/client/packaging/pypi/CHANGELOG.md

All notable future changes to the `delphi_epidata` python client will be documented in this file.
The format is based on [Keep a Changelog](http://keepachangelog.com/).

> ## [4.1.17] - 2024-01-30

Reviewer: you forgot to re-add the link to #1418 and that functionality should be under the pending 4.1.21 heading

All notable future changes to the `delphi_epidata` python client will be documented in this file.
The format is based on [Keep a Changelog](http://keepachangelog.com/).

## [4.1.21] - TBD

### Includes
- https://github.com/cmu-delphi/delphi-epidata/pull/1418

> - https://github.com/cmu-delphi/delphi-epidata/pull/1436
> 
> ### Added
> - Adds two debug flags: `debug` logs info about HTTP requests and responses; whereas `sandbox` prevents any HTTP requests from actually executing, allowing for tests that do not incur server load.
> 
> ### Changed
> - Updates internal version detection to use `bumpversion`.

Reviewer: Lets keep this PR focused on what is already committed; #1436 should make its own updates to the changelog.
```suggestion

### Added
- Adds two debug flags:
  - `debug` logs info about HTTP requests and responses
  - `sandbox` prevents any HTTP requests from actually executing, allowing for tests that do not incur server load.
```

Reviewer: you took out the change from #1418!

Author: added this in again!

## [4.1.17] - 2024-01-30

### Includes
- https://github.com/cmu-delphi/delphi-epidata/pull/1363

> ### Removed
> - Removes uses of setuptools' deprecated pkg_resources library - these are replaced by the native importlib.metadata library.

Reviewer: This was more a replacement than a removal
```suggestion
### Changed
- Replaced use of deprecated setuptools' `pkg_resources` library with the native `importlib.metadata` library.
```

## [4.1.13] - 2023-11-04

### Includes
- https://github.com/cmu-delphi/delphi-epidata/pull/1323
- https://github.com/cmu-delphi/delphi-epidata/pull/1330

### Changed
- Appends a trailing slash to URLs requested by the Python client, which should prevent an automatic redirect and an extra request to the server.

### Removed

> - Mutes the covidcast_nowcast endpoint & removes unnecessary code that is related to this endpoint.

Reviewer: These notes should be phrased as being from the perspective of the client library and things its users will be interested in:
```suggestion
- Removed the `covidcast_nowcast()` method, as the associated API endpoint is no longer available.
```
